modules/core/shared/src/main/scala/lucuma/core/validation/InputValidSplitEpi.scala

```diff
@@ -0,0 +1,181 @@
+      (iso.get _).andThen(_.asRight),
+      iso.reverseGet
+    )
+
+  /**
+   * Build a `InputValidSplitEpi` for `String Refined P`
+   */
+  def refinedString[P](implicit
+    v: RefinedValidate[String, P]
+  ): InputValidSplitEpi[String Refined P] =
+    id.refined[P](NonEmptyChain("Invalid value"))
+
+  /**
+   * `InputValidSplitEpi` for `NonEmptyString`
+   */
+  val nonEmptyString: InputValidSplitEpi[NonEmptyString] =
+    refinedString[NonEmpty].withErrorMessage("Must be defined")
+
+  /**
+   * `InputValidSplitEpi` for `Int`
+   */
+  val int: InputValidSplitEpi[Int] =
+    InputValidSplitEpi(
+      s => fixIntString(s).toIntOption.toRight(NonEmptyChain("Must be an integer")),
+      _.toString
+    )
+
+  /**
+   * Build a `InputValidSplitEpi` for `Int Refined P`
+   */
+  def refinedInt[P](implicit v: RefinedValidate[Int, P]): InputValidSplitEpi[Int Refined P] =
+    int.refined[P](NonEmptyChain("Invalid format"))
+
+  /**
+   * `InputValidSplitEpi` for `PosInt`
+   */
+  val posInt: InputValidSplitEpi[PosInt] =
+    refinedInt[Positive]
+
+  /**
+   * `InputValidSplitEpi` for `BigDecimal`.
+   *
+   * Does not, and cannot, format to a particular number of decimal places. For that you need a
+   * `TruncatedBigDecimal`.
+   */
+  val bigDecimal: InputValidSplitEpi[BigDecimal] =
+    InputValidSplitEpi(
+      s => fixDecimalString(s).toBigDecimalOption.toRight(NonEmptyChain("Must be a number")),
+      _.toString.toLowerCase.replace("+", "") // Strip + sign from exponent.
+    )
+
+  /**
+   * Build a `InputValidSplitEpi` for `BigDecimal Refined P`
+   */
+  def refinedBigDecimal[P](implicit
+    v: RefinedValidate[BigDecimal, P]
+  ): InputValidSplitEpi[BigDecimal Refined P] =
+    bigDecimal.refined[P](NonEmptyChain("Invalid format"))
+
+  /**
+   * `InputValidSplitEpi` for `PosBigDecimal`
+   */
+  val posBigDecimal: InputValidSplitEpi[PosBigDecimal] =
+    refinedBigDecimal[Positive]
+
+  /**
+   * `InputValidSplitEpi` for `BigDecimal`, formatting with only one integer digit.
+   */
+  val bigDecimalWithScientificNotation: InputValidSplitEpi[BigDecimal] =
+    InputValidSplitEpi(
+      bigDecimal.getValid,
+      n => Try(scientificNotationFormat(n)).toOption.orEmpty
+    ) // We shouldn't need to catch errors here, but https://github.com/scala-js/scala-js/issues/4655
+
+  /**
+   * Build a `InputValidSplitEpi` for `BigDecimal Refined P`, formatting with only one integer
+   * digit.
+   */
+  def refinedBigDecimalWithScientificNotation[P](implicit
+    v: RefinedValidate[BigDecimal, P]
+  ): InputValidSplitEpi[BigDecimal Refined P] =
+    bigDecimalWithScientificNotation.refined[P](NonEmptyChain("Invalid format"))
+
+  /**
+   * `InputValidSplitEpi` for `PosBigDecimal`, formatting with only one integer digit.
+   */
+  val posBigDecimalWithScientificNotation: InputValidSplitEpi[PosBigDecimal] =
+    refinedBigDecimalWithScientificNotation[Positive]
+
+  private def fixIntString(str: String): String = str match {
```

Comment:
This is not important but I wonder if these should be in lucuma-ui as I guess they are specific to how we display and edit values?

Reply:
Are you referring to the `fix*` methods? Those are called by the `int` and `bigDecimal` validators.

I moved the predefined numeric and model validators since I'm guessing can be used by the API.